api/example.com/resource/gpu/v1alpha1/sharing_test.go

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2024 The Kubernetes Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1alpha1
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestGpuSharingGetTimeSlicingConfig(t *testing.T) {
+	tests := []struct {
+		name        string
+		gpuSharing  *GpuSharing
+		expected    *TimeSlicingConfig
+		expectedErr string
+	}{
+		{
+			name:        "nil GpuSharing",
+			gpuSharing:  nil,
+			expectedErr: "no sharing set to get config from",
+		},
+		{
+			name: "strategy is not TimeSlicing",
+			gpuSharing: &GpuSharing{
+				Strategy: "not" + TimeSlicingStrategy,
```

Comment:
Question: Does using a valid non-TimeSlicingStrategy here make sense?

Reply:
Yeah that would be equivalent here. I tend toward this kind of mangling of the "correct" value make it "incorrect" to guard against the extreme edge case where two constants like `TimeSlicingStrategy` and `SpacePartitioningStrategy` actually have the same value. Overall though I just want to make these tests as least surprising or interesting as possible, so I can change this.